src/codeflare_sdk/cluster/cluster.py

```diff
@@ -327,7 +333,7 @@ def list_all_queued(namespace: str, print_to_console: bool = True):
 
 def get_current_namespace():  # pragma: no cover
```

Comment:
Note that this function will need to work regardless of auth method

Reply:
I have altered the `config_check` function so that it will now pass the path to the user's specified config file. This will make it so that when `get_current_namespaces()` is called `config.list_kube_config_contexts()` will use the context from the user-defined config file.

With regards to when people authenticate using token and server.
`config.list_kube_config_contexts()` will not work because the user does not have a config file. 
However, when a user authenticates via this method a file is updated with the current namespace `/var/run/secrets/kubernetes.io/serviceaccount/namespace`.

I can add some functionality to read the current namespace from this file if it exists and if it doesn't I could set the namespace to `default` or throw an error to tell the user to specify a namespace themselves.

Reply:
Yes, I think adding something like that would be perfect. If we have any way of telling the user the current namespace being assumed, we should. It really mainly matters because when a `ClusterConfiguration` is created without a specified namespace, it will call `get_current_namespace` to fill that in for the user. So we also need to have appropriate error handling for the case where a user authenticates via server/token, defines a cluster without passing in namespace, and now we either have to gracefully tell them to set it manually (if we cant find it anywhere), or pick a default (maybe `default`). @MichaelClifford Any preference between default being `default` if no way to get namespace, vs printing a warning and returning `None` for `get_current_namespace`, and then a ValueError or something for the `ClusterConfiguration`?

Reply:
Defaulting to "default" I do not think is a real option as the average user likely doesn't actually have permissions to the default namespace. 

When you authenticate to OpenShift you always get associated with some namespace at login. So calling `get_current_namespace()` is always an option. Does kube authentication allow you to login to a cluster, but not a specific namespace?   

If so, I think we should add some error handling to inform the user that they need to specify a namespace.